src/entities/communitySbt.ts

```diff
@@ -174,6 +205,166 @@ class SBT {
         return totalSupply;
     }
 
+    public async getBadgeStatus(args: GetBadgesStatusArgs) : Promise<Array<BadgeWithStatus>| void> {
+        if (this.signer) {
+            const userAddress = await this.signer.getAddress();
+            const network = await this.provider?.getNetwork();
+            const networkName = network ? network.name : "";
+
+            const getURL = this.getEtherscanURL(networkName, args.apiKey, userAddress);
+            const resp = await axios.get(getURL);
+
+            const multiRedeemId = this.getMethodId(true, networkName);
```

Comment:
move this under line 222, no need to call these functions if 220 fails

Reply:
replaced by constants in d6bd73e951951d08b925ccfe8c6a7cb4c91861f6 as they don't depend on the network (my bad)